rust/sbp/src/json/ser.rs

```diff
@@ -194,11 +193,7 @@ impl<F: Formatter + Clone> Encoder<Json2JsonInput> for Json2JsonEncoderInner<F>
     fn encode(&mut self, input: Json2JsonInput, dst: &mut BytesMut) -> Result<(), Self::Error> {
         let formatter = self.formatter.clone();
         let payload = base64::decode(input.data.payload)?;
-        let msg = Sbp::from_frame(Frame {
-            msg_type: input.data.msg_type,
-            sender_id: input.data.sender,
-            payload: BytesMut::from(&payload[..]),
-        })?;
+        let msg = Sbp::from_field(input.data.msg_type, input.data.sender, BytesMut::from(&payload[..]))?;
```

Comment:
something like this might help for that other change I mentioned

```suggestion
        let mut frame = BytesMut::new();
       // push header + payload
       let frame = crate::Frame(frame);
       let msg = Sbp::from_frame(&frame)?;
```